packages/account/src/providers/utils/helpers.ts

```diff
@@ -0,0 +1,29 @@
+type ExcludeResourcesString = {
+  utxos: string[];
+  messages: string[];
+};
+
+export const adjustResourcesToExclude = (params: {
+  userInput: ExcludeResourcesString;
+  cached: ExcludeResourcesString;
+  maxInputs: number;
+}) => {
+  const { userInput, cached, maxInputs } = params;
+
+  const final = { ...userInput };
+
+  let total = final.utxos.length + final.messages.length;
+  if (total >= maxInputs) {
+    return final;
+  }
+
+  final.utxos = [...final.utxos, ...cached.utxos.slice(0, maxInputs - total)];
```

Comment:
```suggestion
  final.utxos = [...final.utxos, ...cached.utxos.reverse().slice(0, maxInputs - total)];
```

The active cache will return in order of the expiry.

| Tx ID | Number of UTXOs | Notes                       |
| ----- | --------------- | --------------------------- |
| 0x01  | 100             | (first transaction submitted)            |
| 0x02  | 100             |                             |
| 0x03  | 100             | <--- We remove these resources from the exclude first |

I am assuming that the earlier the transaction (`0x01`), the higher the likelihood of the transaction being processed by the time we fetch UTXOs/messages (less reason to exclude these resources). Would it be beneficial to reverse the order of excluding? 

| Tx ID | Number of UTXOs | Notes                       |
| ----- | --------------- | --------------------------- |
| 0x01  | 100             | <---  We want to remove these from the exclude first           |
| 0x02  | 100             |                             |
| 0x03  | 100             |  |

Reply:
https://github.com/FuelLabs/fuels-ts/pull/3697/commits/27b02765a77d6cabba134afe5ffe4f71fb29bb64